### PR TITLE
Add auth header to the http call to Token exchange service

### PR DESCRIPTION
**Pull request by smithab**
Things to consider before you submit the PR:

* [ ] Has `CHANGELOG.md` been updated?
* [ ] Are tests passing locally?
* [ ] Are the files formatted correctly?
* [ ] Did you add unit tests?
* [ ] Did you check UI tests on the sample app? They are not executed on CI.
* [ ] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

Update the Token exchange http call to add a authorization header based on what we get from identity

## Related PRs or issues

List related PRs and other issues.

## Misc

Add what's missing, notes on what you tested, additional thoughts or questions.

### CODE REVIEW

File: AppCenterDataStorage/AppCenterDataStorage/Internal/Client/MSTokenExchange.m [hunk 1]
```diff
@@ -34,6 +36,13 @@ + (void)performDbTokenAsyncOperationWithHttpClient:(MSStorageIngestion *)httpCli
     NSData *payloadData = [NSJSONSerialization dataWithJSONObject:@{kMSPartitions : @[ partition ]} options:0 error:&jsonError];
 
     // Http call.
+    if ([MSAuthTokenContext sharedInstance].authToken) {
+      NSMutableDictionary *headers = [httpClient.httpHeaders mutableCopy];
```
Comment (jaelim-ms) [inline AppCenterDataStorage/AppCenterDataStorage/Internal/Client/MSTokenExchange.m]: Is `httpHeaders` immutable?

Comment (dhei) [reply AppCenterDataStorage/AppCenterDataStorage/Internal/Client/MSTokenExchange.m]: @jaelim-ms `httpHeaders` is `NSDictionary` that is immutable.

File: AppCenterDataStorage/AppCenterDataStorage/Internal/Client/MSTokenExchange.m [hunk 2]
```diff
@@ -34,6 +36,13 @@ + (void)performDbTokenAsyncOperationWithHttpClient:(MSStorageIngestion *)httpCli
...
+      NSString *bearerTokenHeader = [NSString stringWithFormat:kMSBearerTokenHeaderFormat, [MSAuthTokenContext sharedInstance].authToken];
```
Comment (achocron) [inline AppCenterDataStorage/AppCenterDataStorage/Internal/Client/MSTokenExchange.m]: nit: this probably doesn't need to be stored in a variable

File: AppCenterDataStorage/AppCenterDataStorage/Internal/Client/MSTokenExchange.m [hunk 3]
```diff
@@ -34,6 +36,12 @@ + (void)performDbTokenAsyncOperationWithHttpClient:(MSStorageIngestion *)httpCli
     NSData *payloadData = [NSJSONSerialization dataWithJSONObject:@{kMSPartitions : @[ partition ]} options:0 error:&jsonError];
 
     // Http call.
```
Comment (ElektrojungeAtWork) [inline AppCenterDataStorage/AppCenterDataStorage/Internal/Client/MSTokenExchange.m]: This comment seems unnecessary to me. The variable before it is called `payloadData` and the logic below is like 10 lines of self-explanatory code.

**Pull request merged**